File: javascript/collatz-conjecture/Iteration1/collatz-conjecture.js

export const steps = (number) => {
    if (number <= 0) {
        throw 'Only positive numbers are allowed'
    }

    var numberOfSteps = 0;
    var result = number

    while (result !== 1) {
        
        if (isEven(result)) {
            result = result / 2;
        }
        else {
            result = (3 * result) + 1;
        }

        numberOfSteps++;
    }

    return numberOfSteps;
};

function isEven(number) {
    return (number % 2) === 0;
}

File: nim/collatz-conjecture/Iteration1/collatz_conjecture.nim
proc steps*(n: int): int =
  if (n <= 0):
    raise newException(ValueError, "Number cannot be negative or zero")

  var number = n

  while number > 1:
    if number mod 2 == 0:
      number = int(number / 2)
    else:
      number = (number * 3) + 1

    inc(result)